### PR TITLE
[KOGITO-261] Installs default Data Index service when creating a new …

### PR DESCRIPTION
**Pull request by ricardozanini**
…project

See:
https://issues.redhat.com/browse/KOGITO-261

Installs Data Index upon a new project creation. Depends on #118.

After merging #118 we will install the `KogitoInfra` as well to have a way to have a full Data Index experience once the user creates the new project.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: cmd/kogito/command/project/new_project.go [hunk 1]
```diff
@@ -74,6 +76,7 @@ func (i *newProjectCommand) InitHook() {
 	i.flags = newProjectFlags{}
 	i.Parent.AddCommand(i.command)
 	i.command.Flags().StringVarP(&i.flags.project, "project", "n", "", "The project project")
+	i.command.Flags().BoolVar(&i.flags.installDataIndex, "install-data-index", true, "Installs the default instance of Data Index being provisioned by the Kogito Operator in the new project")
```
Comment (radtriste) [inline cmd/kogito/command/project/new_project.go]: 'true' means it will be installed by default ?

Comment (ricardozanini) [reply cmd/kogito/command/project/new_project.go]: You and @mswiderski could start a fight regarding this flag :p
I just want to merge this PR by the end of the day to have time to push it to 0.6.0 :)

**Comment by ricardozanini** [comment]
We will wait for https://issues.redhat.com/browse/KOGITO-703 to be fixed to implement this one.

**Comment by radtriste** [comment]
LGTM. Tests are ok. Will implement a Cucumber test on CLI as part of Kogito-491

File: pkg/controller/kogitodataindex/kogitodataindex_controller.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func (r *ReconcileKogitoDataIndex) Reconcile(request reconcile.Request) (reconci
 	// Deploy infra dependencies
 	if result, err := r.ensureKogitoInfra(instance); err != nil {
 		return reconcile.Result{}, err
-	} else if result != nil {
+	} else if result != nil && result.RequeueAfter > 0 {
```
Comment (xiezhang7) [inline pkg/controller/kogitodataindex/kogitodataindex_controller.go]: Updating the `instance` in `ensureKogitoInfra` is supposed to trigger a new iteration of `reconcile`.
https://github.com/kiegroup/kogito-cloud-operator/blob/ac0d2dbbc82d6dc310736ebc5a55c07e8cd95e63/pkg/controller/kogitodataindex/kogitodataindex_controller.go#L191
In this case, the current iteration of `reconcile` should return without requeue, and let the new iteration of `reconcile` to handle changes.
I also created a fix to this issue in pull request #133 . https://github.com/kiegroup/kogito-cloud-operator/blob/1ca5a3b3ed9c77f331405bd7aa556d1a34d3bb79/pkg/controller/kogitodataindex/kogitodataindex_controller.go#L232

Comment (ricardozanini) [reply pkg/controller/kogitodataindex/kogitodataindex_controller.go]: Yes, you're right. The trick part was the `return nil, nil`. Thanks! Updated.

**Pull request merged**